src/lib/utils/thorchain/index.ts

```diff
@@ -40,7 +40,15 @@ const getThorchainTransactionStatus = async (txHash: string, skipOutbound?: bool
     (!skipOutbound && thorTxData.stages.outbound_signed?.completed === false)
   )
     return TxStatus.Pending
-  if (thorTxData.stages.swap_status?.pending === false) return TxStatus.Confirmed
+  if (
+    // Skips outbound checks. If the swap is complete, we assume the transaction as confirmed
+    (skipOutbound && thorTxData.stages.swap_status?.pending === false) ||
+    // When enforcing outbound checks, ensures the outbound Tx is signed and an out Tx is present
+    (thorTxData.stages.outbound_signed?.completed && thorTxData.out_txs)
+  ) {
+    // TODO(gomes): introspect thornode and handle failed refunds
```

Comment:
Discussed this one with @kaladinlight today, will add a follow-up to that to handle failed refunds.
We'll effectively need to introspect midgard in conjunction with the status endpoint, as well as change the implementation/consumption of this guy to handle failures as well, which it doesn't currently

See e.g this Tx with the status endpoint seemingly succesful:

https://dev-daemon.thorchain.shapeshift.com/lcd/thorchain/tx/status/180d89494f9ae1c28a0647c3ae6b5a4f625ff622ecb822fa85d21236c8580058

but actually a failed refund: https://dev-daemon.thorchain.shapeshift.com/lcd/thorchain/tx/status/32C093783A806767A2CE92D735A1CFEC67D0BE9916E5EC510BC2101D6A8EEB23

Since the refund failed, there was no refund outbound Tx, which means that the status endpoint cannot be used to detect failed refunds.